Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -2873,6 +2874,11 @@ public void init() {
 
             Activity activity = branch.getCurrentActivity();
             Intent intent = activity != null ? activity.getIntent() : null;
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1 && activity.getReferrer() != null) {
```

Comment:
@msiddiq-branch I'm assuming `Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1` is OK by product? Or if not, is there any other way we can achieve the same functionality on the lower API levels?

Also, nitpick: let's use `!TextUtils.isEmpty(activity.getReferrer())` instead of `activity.getReferrer() != null`, that way we'll capture empty strings as well as nulls.

Finally, this reminded me of a recent inquiry about the intent keys that Branch uses, this would have potentially gone under the radar because the intent keys we use to extract data via `getReferrer()` are not listed among `Defines.IntentKeys`.. so maybe we could add a comment ssomewhere e inside `Defines.IntentKeys` that says:

```
    // The below intent keys are also used to extract data from the intent (via Activity.getReferrer())
    // public static final String EXTRA_REFERRER = "android.intent.extra.REFERRER";
    // public static final String EXTRA_REFERRER_NAME = "android.intent.extra.REFERRER_NAME";
```

Reply:
@bklastaitis-branch Currently since our SDK supports minimum SDK as 21 I believe we need to pass this value from API 21. I will figure out a method to fetch this value in the lower API version. Currently, getReferrer is supported from Api 22. cc @aaaronlopez

Reply:
@bklastaitis-branch I have replaced activity.getReferrer with Activitycompat.getReferrer but still we are able to get referrer info only from API version 22 ideally this should have worked. I have been testing this on an emulator I'm not sure if that could be the reason. As per the code from google the implementation for version lower than 22 would something as mentioned [here](https://github.com/googlecodelabs/deeplink-referrer/blob/4c3f7bde356fec47d67f83c19c07fabe1cb01bd5/end/app/src/main/java/com/google/samples/search/recipe_app/client/AnalyticsApplication.java#L129)  and Activitycompat.getReferrer has the same implementation.

@aaaronlopez Are we okay to have this supported from API 22?

Reply:
min api support being LOLLIPOP_MR1 is fine by me for this API 👍

Reply:
Thanks for confirming Aaron.